resip/stack/ParserContainer.hxx

```diff
@@ -177,10 +177,8 @@ class ParserContainer : public ParserContainerBase
             iterator operator++(int) {iterator it(mIt++,mRef); return it;}
             iterator operator--() {iterator it(--mIt,mRef); return it;}
             iterator operator--(int) {iterator it(mIt--,mRef); return it;}
-            bool operator!=(const iterator& rhs) { return mIt != rhs.mIt; }
-            bool operator==(const iterator& rhs) { return mIt == rhs.mIt; }
-            bool operator!=(const const_iterator& rhs) { return mIt != rhs.mIt; }
-            bool operator==(const const_iterator& rhs) { return mIt == rhs.mIt; }
+            friend bool operator!=(const iterator& lhs, const iterator& rhs) { return lhs.mIt != rhs.mIt; }
+            friend bool operator==(const iterator& lhs, const iterator& rhs) { return lhs.mIt == rhs.mIt; }
```

Comment:
@miscco The (non-const) iterator and iterator should be comparable. But I struggle to provide a C++11 - 20 compatible solution without creating a mess. Do you have an idea for a minimal-invasive change?

Reply:
I think the issue might be the rewriting operator.

I think this should work: https://godbolt.org/z/zrsTnW9YE